### PR TITLE
Reference Docs

### LINKED ISSUE #846: <py-script> tag needs reference docs
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a duplication
- [X] I already searched in Google and didn't find any good information or help

### What is the issue/comment/problem?

The `<py-script>` tag needs a reference doc that explains what the tag is for and any attributes related to it.

The doc file should be at `source/reference/py-script.md` and referenced in `source/reference/index.md`.

### LINKED ISSUE #847: <py-config> tag needs reference docs
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a duplication
- [X] I already searched in Google and didn't find any good information or help

### What is the issue/comment/problem?

The `<py-config>` tag needs a reference doc that explains what the tag is for and any attributes related to it.

The doc file should be at `source/reference/py-config.md` and referenced in `source/reference/index.md`.

### LINKED ISSUE #849: <py-repl> needs reference docs
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a duplication
- [X] I already searched in Google and didn't find any good information or help

### What is the issue/comment/problem?

The `<py-repl>` tag needs a reference doc that explains what the tag is for and any attributes related to it.

The doc file should be at `source/reference/py-repl.md` and referenced in `source/reference/index.md`.

### PR DESCRIPTION
**Pull request by tedpatrick**
Add Reference Elements & Modules to docs for `<py-script>` `<py-config>` `<py-repl>`

Fixes #847 
Fixes #849 
Fixes #846

### CODE REVIEW

File: docs/reference/elements/py-config.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    "packages": ["./static/wheels/travertino-0.1.3-py3-none-any.whl"]
+  }
+</py-config>
+```
+
+If your `.whl` is not a pure Python wheel, then open a PR or issue with [pyodide](https://github.com/pyodide/pyodide) to get it added [here](https://github.com/pyodide/pyodide/tree/main/packages).
+
+If there's enough popular demand, the pyodide team will likely work on supporting your package. Regardless, things will likely move faster if you make the PR and consult with the team to get unblocked.
+
+For example, NumPy and Matplotlib are available. Notice here we're using `<py-script output="plot">`
```
Comment (fpliger) [inline docs/reference/elements/py-config.md:123]: Very minor comment... For readability, I'd move the "Notice here... " after the example. Maybe a personal thing though...

File: docs/reference/elements/py-config.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+Note: While the `<py-config>` tag supports both TOML and JSON, one cannot mix the type of config passed from 2 different sources i.e. the case when inline config is in TOML format while config from src is in JSON format is NOT allowed. Similarly for the opposite case.
+
+---
+
+This is helpful in cases where a number of applications share a common configuration (which can be supplied via `src`), but their specific keys need to be customised and overridden.
+
+The keys supplied through `inline` override the values present in config supplied via `src`.
+
+One can also declare dependencies so as to get access to many 3rd party OSS packages that are supported by PyScript.
+You can also link to `.whl` files directly on disk like in our [toga example](https://github.com/pyscript/pyscript/blob/main/examples/toga/freedom.html).
```
Comment (fpliger) [inline docs/reference/elements/py-config.md:101]: I'd add something like:

Package dependencies in the config can be declared by using the direct link to the package URL (whl or any other format supported by the chosen runtime) or by just providing the package name [and version]. If only the name [and version] are provided, packages will be installed directly from what's provided by your runtime or from PyPI. NOTICE that only pure python packages from PyPI will work and packages with C dependencies will not. These need to be built specifically for WASM (please, consult the Pyodide project for more information about what's supported and on how to build packages with c dependencies)

File: docs/reference/elements/py-config.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+OR in JSON like
+
+```
+<py-config type="json">
+  {
+    "magic": "unicorn"
+  }
+</py-config>
+```
+
```
Comment (fpliger) [inline docs/reference/elements/py-config.md:241]: I think it'd be extremely valuable to users if we provided an example of a full configuration with all the values available.

File: docs/reference/elements/py-config.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+| `runtimes` | List of Runtimes | List of runtime configurations, described below. The default value contains a single Pyodide based runtime. |
+
+A runtime configuration consists of the following:
+| Value | Type | Description |
+| ----- | ---- | ----------- |
+| `src` | string (Required) | URL to the runtime source. |
+| `name` | string | Name of the runtime. This field can be any string and is to be used by the application author for their own customization purposes |
+| `lang` | string | Programming language supported by the runtime. This field can be used by the application author to provide clarification. It currently has no implications on how PyScript behaves. |
+
+Besides the above format, a user can also supply any extra keys and values that are relevant as metadata information or perhaps are being used within the application.
```
Comment (fpliger) [inline docs/reference/elements/py-config.md:222]: format -> schema ?

File: docs/reference/elements/py-repl.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# &lt;py-repl&gt;
+
+The `<py-repl>` element provides a REPL(Read Eval Print Loop) to evaluate multi-line Python and display output.
+
+## Attributes
+
+<<<<<<< HEAD
```
Comment (fpliger) [inline docs/reference/elements/py-repl.md]: left over?

**Review by fpliger — APPROVED** [review]
Thank you @tedpatrick !! THis is an excellent PR! Left some comments but overall lgtm

File: docs/reference/elements/py-config.md (hunk 2 above)
Comment (tedpatrick) [reply docs/reference/elements/py-config.md:101]: added

File: docs/reference/elements/py-config.md (hunk 3 above)
Comment (tedpatrick) [reply docs/reference/elements/py-config.md:245]: Added complete schema in json + toml as examples

**Pull request merged**